x-pack/plugins/synthetics/common/constants/monitor_management.ts

```diff
@@ -99,12 +88,22 @@ export const secretKeys = [
   ConfigKey.RESPONSE_RECEIVE_CHECK,
   ConfigKey.SOURCE_INLINE,
   ConfigKey.SOURCE_PROJECT_CONTENT,
-  ConfigKey.SOURCE_ZIP_USERNAME,
-  ConfigKey.SOURCE_ZIP_PASSWORD,
   ConfigKey.SYNTHETICS_ARGS,
   ConfigKey.TLS_KEY,
   ConfigKey.TLS_KEY_PASSPHRASE,
   ConfigKey.USERNAME,
-  ConfigKey.ZIP_URL_TLS_KEY,
-  ConfigKey.ZIP_URL_TLS_KEY_PASSPHRASE,
 ] as const;
+
+export enum LegacyConfigKey {
```

Comment:
@shahzad31 For the throttling feature, any keys you want to remove from the saved object you'll want to make sure you include in this enum. This will ensure that the old legacy fields are excluded from AAD when migrated, and ensure that decryption is able to progress with the legacy fields missing.